testframework/src/main/java/net/neoforged/testframework/gametest/GameTestHelperFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.testframework.gametest;
+
+import java.lang.invoke.MethodType;
+import java.util.function.Function;
+import net.minecraft.gametest.framework.GameTestHelper;
+import net.minecraft.gametest.framework.GameTestInfo;
+import net.neoforged.testframework.impl.ReflectionUtils;
+
+public interface GameTestHelperFactory<T extends GameTestHelper> extends Function<GameTestInfo, T> {
```

Comment:
This can be in the impl package.

Reply:
Done.

Reply:
Also would it be better to expose this interface and make the `void onGameTest(GameTestHelperFactory<T> factory, final Consumer<T> consumer)` method public?